client/web/src/components/admin-panel-link.tsx

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+import React, { useMemo } from "react"
+import { NodeData } from "src/hooks/node-data"
+
+/**
+ * AdminPanelLink displays a link to the node's control server admin panel,
+ * with the specified suffix path. If the node's control server doesn't have
+ * an admin panel, no content is rendered.
+ *
+ * The link contents should be provided as a child component,
+ * with the desired clickable text wrapped between [] brackets.
+ * e.g. "Visit the [admin webhooks page] to learn more."
+ */
+export default function AdminPanelLink({
```

Comment:
Might have over-engineered this a little by adding a new component, just didn't feel like repeating the control server logic at all the various spots.

Reply:
I actually like having a new component to capture this logic, but am not sure about the `[link]` parsing 😕 .  I mean, it's not likely that would would have square brackets in the text, but it seems a little odd versus just having the link be more explicit.  Would it be terrible to have this return two components, one for the wrapper that has the conditional logic, and one for the link itself which adds `target="_blank" rel="noreferrer"` and such?

I also kinda wish we had swr so that AdminPanelLink could fetch nodeData directly (without an actual roundtrip) and you didn't have to pass the whole object in.

Reply:
> I also kinda wish we had swr so that AdminPanelLink could fetch nodeData directly (without an actual roundtrip) and you didn't have to pass the whole object in.

Yeaaaaa

> but am not sure about the [link] parsing 😕

One other thing I was considering was just adding "pre" and "post" props, so it's all explicit but still one component. I didn't love having to split out the child contents, but I agree it's a little hacky in curent form.

Reply:
Okay actually I don't hate the two components thing:
```
      <AdminLink.Container
        className="text-neutral-500 text-sm leading-tight"
        node={node}
      >
        Remember to make sure that the{" "}
        <AdminLink.Link node={node} path="/acls">
          tailnet policy file
        </AdminLink.Link>{" "}
        allows other devices to SSH into this device.
      </AdminLink.Container>
    </>
```
Will update

Reply:
Updated, ptal